Marketplace/src/main/java/com/marketplace/service/category/JdbcCategoryService.java

```diff
@@ -0,0 +1,145 @@
+package com.marketplace.service.category;
+
+import com.marketplace.config.ImageLoader;
+import com.marketplace.repository.category.CategoryRepository;
+import com.marketplace.repository.category.DbCategory;
+import com.marketplace.repository.category.DbCharacteristic;
+import com.marketplace.repository.category.ProductQuery;
+import com.marketplace.config.exception.NonExistingEntityException;
+import com.marketplace.config.exception.ParentCategoryException;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class JdbcCategoryService implements CategoryService {
```

Comment:
Service name shouldn't contain db-related words. As non specific implementation you can use CategoryServiceImpl name